diff --git a/rosidl_typesupport_microxrcedds_cpp/CMakeLists.txt b/rosidl_typesupport_microxrcedds_cpp/CMakeLists.txt
index 36a97d7..56a9ff5 100644
--- a/rosidl_typesupport_microxrcedds_cpp/CMakeLists.txt
+++ b/rosidl_typesupport_microxrcedds_cpp/CMakeLists.txt
@@ -57,7 +57,6 @@ target_compile_options(${PROJECT_NAME}
   PRIVATE
     $<$<C_COMPILER_ID:GNU>:-Wall>
     $<$<C_COMPILER_ID:GNU>:-Wextra>
-    $<$<C_COMPILER_ID:GNU>:-Wpedantic>
   )
 
 set_target_properties(${PROJECT_NAME} PROPERTIES
