diff --git a/src/c/util/time.c b/src/c/util/time.c
index c4d9de6..0b69c77 100644
--- a/src/c/util/time.c
+++ b/src/c/util/time.c
@@ -7,8 +7,8 @@
 #elif defined(UCLIENT_PLATFORM_FREERTOS_PLUS_TCP)
 #include "FreeRTOS.h"
 #include "task.h"
-#elif defined(UCLIENT_PLATFORM_ZEPHYR)
-#include <version.h>
+//#elif defined(UCLIENT_PLATFORM_ZEPHYR)
+//#include <version.h>
 #endif /* ifdef WIN32 */
 
 //==================================================================
@@ -60,11 +60,11 @@ int64_t uxr_nanos(
     // has been renamed to the clock_gettime function.
     // This has been done to implement Zephyr's POSIX API as regular library functions
     // https://github.com/zephyrproject-rtos/zephyr/commit/95a22b12174621aeba8ca3e0e61f7c66f03202bf
-    #if ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 5, 99)
+    //#if ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 5, 99)
     clock_gettime(CLOCK_REALTIME, &ts);
-    #else
-    z_impl_clock_gettime(CLOCK_REALTIME, &ts);
-    #endif /* if ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 5, 99) */
+    //#else
+    //z_impl_clock_gettime(CLOCK_REALTIME, &ts);
+    //#endif /* if ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 5, 99) */
     return (((int64_t)ts.tv_sec) * 1000000000) + ts.tv_nsec;
 #else
     struct timespec ts;
