diff --git a/config/zephyr/generic/client-colcon.meta b/config/zephyr/generic/client-colcon.meta
index 3a76c7b..1462ee1 100644
--- a/config/zephyr/generic/client-colcon.meta
+++ b/config/zephyr/generic/client-colcon.meta
@@ -2,17 +2,23 @@
     "names": {
         "tracetools": {
             "cmake-args": [
+                "-DBUILD_SHARED_LIBS=OFF",
+                "-DENABLE_TESTING=OFF",
                 "-DTRACETOOLS_DISABLED=ON",
                 "-DTRACETOOLS_STATUS_CHECKING_TOOL=OFF"
             ]
         },
         "rosidl_typesupport": {
             "cmake-args": [
+                "-DBUILD_SHARED_LIBS=OFF",
+                "-DENABLE_TESTING=OFF",
                 "-DROSIDL_TYPESUPPORT_SINGLE_TYPESUPPORT=ON"
+                "-DENABLE_TESTING=OFF",
             ]
         },
         "rcl": {
             "cmake-args": [
+                "-DBUILD_SHARED_LIBS=OFF",
                 "-DBUILD_TESTING=OFF",
                 "-DRCL_COMMAND_LINE_ENABLED=OFF",
                 "-DRCL_LOGGING_ENABLED=OFF"
@@ -20,7 +26,9 @@
         }, 
         "rcutils": {
             "cmake-args": [
+                "-DBUILD_SHARED_LIBS=OFF",
                 "-DENABLE_TESTING=OFF",
+                "-DBUILD_TESTING=OFF",
                 "-DRCUTILS_NO_FILESYSTEM=ON",
                 "-DRCUTILS_NO_THREAD_SUPPORT=ON",
                 "-DRCUTILS_AVOID_DYNAMIC_ALLOCATION=ON",
@@ -29,12 +37,16 @@
         },
         "microxrcedds_client": {
             "cmake-args": [
+                "-DBUILD_SHARED_LIBS=OFF",
+                "-DENABLE_TESTING=OFF",
                 "-DUCLIENT_PIC=OFF",
                 "-DUCLIENT_PROFILE_DISCOVERY=OFF"
             ]
         },
         "rmw_microxrcedds": {
             "cmake-args": [
+                "-DBUILD_SHARED_LIBS=OFF",
+                "-DENABLE_TESTING=OFF",
                 "-DRMW_UXRCE_MAX_NODES=1",
                 "-DRMW_UXRCE_MAX_PUBLISHERS=1",
                 "-DRMW_UXRCE_MAX_SUBSCRIPTIONS=1",
