diff --git a/rmw_microxrcedds_c/CMakeLists.txt b/rmw_microxrcedds_c/CMakeLists.txt
index 451719b..b73f2a7 100644
--- a/rmw_microxrcedds_c/CMakeLists.txt
+++ b/rmw_microxrcedds_c/CMakeLists.txt
@@ -305,7 +305,6 @@ target_compile_options(${PROJECT_NAME}
   PRIVATE
     $<$<C_COMPILER_ID:GNU>:-Wall>
     $<$<C_COMPILER_ID:GNU>:-Wextra>
-    $<$<C_COMPILER_ID:GNU>:-pedantic>
     $<$<C_COMPILER_ID:GNU>:-Werror-implicit-function-declaration>
     $<$<C_COMPILER_ID:GNU>:$<$<VERSION_GREATER:$<C_COMPILER_VERSION>,8.1>:-Wcast-align=strict>>
     $<$<C_COMPILER_ID:GNU>:-Wvla>
