diff --git a/rmw_microxrcedds_c/src/rmw_init.c b/rmw_microxrcedds_c/src/rmw_init.c
index 8312173..72b1164 100644
--- a/rmw_microxrcedds_c/src/rmw_init.c
+++ b/rmw_microxrcedds_c/src/rmw_init.c
@@ -34,6 +34,9 @@
 #endif  // RMW_UXRCE_GRAPH
 
 extern rmw_uxrce_transport_params_t rmw_uxrce_transport_default_params;
+void srandom (unsigned int seed);
+long int random (void);
+
 
 rmw_ret_t
 rmw_init_options_init(
@@ -109,10 +112,11 @@ rmw_init_options_init(
   init_options->impl->transport_params.read_cb = rmw_uxrce_transport_default_params.read_cb;
 #endif /* if defined(RMW_UXRCE_TRANSPORT_SERIAL) */
 
-  srand(uxr_nanos());
+  srandom(uxr_nanos());
 
   do {
-    init_options->impl->transport_params.client_key = rand(); //NOLINT
+
+    init_options->impl->transport_params.client_key = random(); //NOLINT
   } while (init_options->impl->transport_params.client_key == 0);
 
   return RMW_RET_OK;
