diff --git a/rcl_logging_interface/CMakeLists.txt b/rcl_logging_interface/CMakeLists.txt
index 6c67659..0600476 100644
--- a/rcl_logging_interface/CMakeLists.txt
+++ b/rcl_logging_interface/CMakeLists.txt
@@ -13,7 +13,7 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 
 if(NOT WIN32)
-  add_compile_options(-Wall -Wextra -Wpedantic)
+  add_compile_options(-Wall -Wextra)
 endif()
 
 find_package(ament_cmake_ros REQUIRED)
