diff --git a/src/time_unix.c b/src/time_unix.c
index 92ada57..d7a8c0c 100644
--- a/src/time_unix.c
+++ b/src/time_unix.c
@@ -25,12 +25,12 @@
 #include <math.h>
 
 #if defined(__ZEPHYR__)
-#include <version.h>
-#if ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 1, 0)
+//#include <version.h>
+//#if ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 1, 0)
 #include <zephyr/posix/time.h>  //  Points to Zephyr toolchain posix time implementation
-#else
-#include <posix/time.h>  //  Points to Zephyr toolchain posix time implementation
-#endif
+//#else
+//#include <posix/time.h>  //  Points to Zephyr toolchain posix time implementation
+//#endif
 #else  //  #if KERNELVERSION >= ZEPHYR_VERSION(3, 1, 0)
 #include <time.h>
 #endif  //  defined(__ZEPHYR__)
