diff --git a/src/logging.c b/src/logging.c
index 186ba24..1a1021b 100644
--- a/src/logging.c
+++ b/src/logging.c
@@ -150,6 +150,7 @@ static enum rcutils_get_env_retval rcutils_get_env_var_zero_or_one(
   const char * one_semantic)
 {
   const char * env_var_value = NULL;
+  (void)zero_semantic;
   const char * ret_str = rcutils_get_env(name, &env_var_value);
   if (NULL != ret_str) {
     RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
