diff --git a/include/zephyr/net/hostname.h b/include/zephyr/net/hostname.h
index e34c1df31a..6398249edc 100644
--- a/include/zephyr/net/hostname.h
+++ b/include/zephyr/net/hostname.h
@@ -103,6 +103,8 @@ int net_hostname_set_postfix(const uint8_t *hostname_postfix,
 static inline int net_hostname_set_postfix(const uint8_t *hostname_postfix,
 					   int postfix_len)
 {
+	(void) hostname_postfix;
+	(void) postfix_len;
 	return -EMSGSIZE;
 }
 #endif /* CONFIG_NET_HOSTNAME_UNIQUE */
