diff --git a/include/zephyr/net/buf.h b/include/zephyr/net/buf.h
index c39f3dd561..53f08c3b6a 100644
--- a/include/zephyr/net/buf.h
+++ b/include/zephyr/net/buf.h
@@ -864,7 +864,7 @@ struct net_buf_simple_state {
 static inline void net_buf_simple_save(struct net_buf_simple *buf,
 				       struct net_buf_simple_state *state)
 {
-	state->offset = net_buf_simple_headroom(buf);
+        state->offset = (uint16_t)net_buf_simple_headroom(buf);
 	state->len = buf->len;
 }
 
