diff --git a/include/zephyr/sys/byteorder.h b/include/zephyr/sys/byteorder.h
index 978255fd02..35aed12e0d 100644
--- a/include/zephyr/sys/byteorder.h
+++ b/include/zephyr/sys/byteorder.h
@@ -307,8 +307,8 @@
  */
 static inline void sys_put_be16(uint16_t val, uint8_t dst[2])
 {
-	dst[0] = val >> 8;
-	dst[1] = val;
+        dst[0] = (uint8_t)(val >> 8);
+	dst[1] = (uint8_t)val;
 }
 
 /**
@@ -322,8 +322,8 @@ static inline void sys_put_be16(uint16_t val, uint8_t dst[2])
  */
 static inline void sys_put_be24(uint32_t val, uint8_t dst[3])
 {
-	dst[0] = val >> 16;
-	sys_put_be16(val, &dst[1]);
+        dst[0] = (uint8_t)(val >> 16);
+	sys_put_be16((uint16_t)val, &dst[1]);
 }
 
 /**
@@ -337,8 +337,8 @@ static inline void sys_put_be24(uint32_t val, uint8_t dst[3])
  */
 static inline void sys_put_be32(uint32_t val, uint8_t dst[4])
 {
-	sys_put_be16(val >> 16, dst);
-	sys_put_be16(val, &dst[2]);
+        sys_put_be16((uint16_t)(val >> 16), dst);
+	sys_put_be16((uint16_t)val, &dst[2]);
 }
 
 /**
@@ -352,8 +352,8 @@ static inline void sys_put_be32(uint32_t val, uint8_t dst[4])
  */
 static inline void sys_put_be48(uint64_t val, uint8_t dst[6])
 {
-	sys_put_be16(val >> 32, dst);
-	sys_put_be32(val, &dst[2]);
+        sys_put_be16((uint16_t)(val >> 32), dst);
+	sys_put_be32((uint32_t)val, &dst[2]);
 }
 
 /**
@@ -367,8 +367,8 @@ static inline void sys_put_be48(uint64_t val, uint8_t dst[6])
  */
 static inline void sys_put_be64(uint64_t val, uint8_t dst[8])
 {
-	sys_put_be32(val >> 32, dst);
-	sys_put_be32(val, &dst[4]);
+        sys_put_be32((uint32_t)(val >> 32), dst);
+	sys_put_be32((uint32_t)val, &dst[4]);
 }
 
 /**
@@ -382,8 +382,8 @@ static inline void sys_put_be64(uint64_t val, uint8_t dst[8])
  */
 static inline void sys_put_le16(uint16_t val, uint8_t dst[2])
 {
-	dst[0] = val;
-	dst[1] = val >> 8;
+        dst[0] = (uint8_t)val;
+	dst[1] = (uint8_t)(val >> 8);
 }
 
 /**
@@ -397,8 +397,8 @@ static inline void sys_put_le16(uint16_t val, uint8_t dst[2])
  */
 static inline void sys_put_le24(uint32_t val, uint8_t dst[3])
 {
-	sys_put_le16(val, dst);
-	dst[2] = val >> 16;
+        sys_put_le16((uint16_t)val, dst);
+	dst[2] = (uint8_t)(val >> 16);
 }
 
 /**
@@ -412,8 +412,8 @@ static inline void sys_put_le24(uint32_t val, uint8_t dst[3])
  */
 static inline void sys_put_le32(uint32_t val, uint8_t dst[4])
 {
-	sys_put_le16(val, dst);
-	sys_put_le16(val >> 16, &dst[2]);
+        sys_put_le16((uint16_t)val, dst);
+	sys_put_le16((uint16_t)(val >> 16), &dst[2]);
 }
 
 /**
@@ -427,8 +427,8 @@ static inline void sys_put_le32(uint32_t val, uint8_t dst[4])
  */
 static inline void sys_put_le48(uint64_t val, uint8_t dst[6])
 {
-	sys_put_le32(val, dst);
-	sys_put_le16(val >> 32, &dst[4]);
+        sys_put_le32((uint32_t)val, dst);
+	sys_put_le16((uint16_t)(val >> 32), &dst[4]);
 }
 
 /**
@@ -442,8 +442,8 @@ static inline void sys_put_le48(uint64_t val, uint8_t dst[6])
  */
 static inline void sys_put_le64(uint64_t val, uint8_t dst[8])
 {
-	sys_put_le32(val, dst);
-	sys_put_le32(val >> 32, &dst[4]);
+        sys_put_le32((uint32_t)val, dst);
+	sys_put_le32((uint32_t)(val >> 32), &dst[4]);
 }
 
 /**
