diff --git a/src/time_unix.c b/src/time_unix.c
index 8cbc40a..9355627 100644
--- a/src/time_unix.c
+++ b/src/time_unix.c
@@ -30,12 +30,12 @@ extern "C"
 #include <math.h>
 
 #if defined(__ZEPHYR__)
-#include <version.h>
-#if ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 1, 0)
+//#include <version.h>
+//#if ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 1, 0)
 #include <zephyr/posix/time.h>  //  Points to Zephyr toolchain posix time implementation
-#else
-#include <time.h>
-#endif // ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 1, 0)
+//#else
+//#include <time.h>
+//#endif // ZEPHYR_VERSION_CODE >= ZEPHYR_VERSION(3, 1, 0)
 #else
 #include <time.h>
 #endif  //  defined(__ZEPHYR__)
